sale_team_invoice_partner/sale_view.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+        <record id="view_order_form" model="ir.ui.view">
+            <field name="model">sale.order</field>
+            <field name="inherit_id" ref="sale.view_order_form"/>
+            <field name="arch" type="xml">
+                <field name="partner_invoice_id" position="after">
+                    <field name="team_partner_invoice_id" invisible="1"/>
+                </field>
+                <field name="partner_invoice_id" position="attributes">
+                    <attribute name="attrs">{'readonly':[('section_partner_invoice_id','!=', False)]}</attribute>
```

Comment:
Do you really want to forbid changing the partner?
Did you mean `team` instead of `section`?